pdns/dnsdistdist/dnsdist-rules.hh

```diff
@@ -991,3 +991,25 @@ private:
   boost::optional<std::string> d_value;
   std::string d_tag;
 };
+
+class PoolAvailableRule : public DNSRule
+{
+public:
+  PoolAvailableRule(pools_t& pools, const std::string& poolname) : d_poolname(poolname)
+  {
+    d_pool = getPool(pools, poolname);
+  }
+
+  bool matches(const DNSQuestion* dq) const override
+  {
+    return (d_pool->countServers(true) > 0);
+  }
+
+  string toString() const override
+  {
+    return "pool '" + d_poolname + "' is available";
+  }
+private:
+  std::string d_poolname;
+  std::shared_ptr<ServerPool> d_pool;
```

Comment:
I'm not sure storing a pointer to the pool is a good idea. I don't see any issue right now because there is currently no way to destroy a pool in `dnsdist` but if we ever implement that, you could be referencing a stale pool (not too bad) or the previous version of a pool that has been destroyed then created again.
The alternative is keeping around a `LocalStateHolder<pools_t>` or getting a new one for each query. The first option should not be much more expensive than the current code.

Reply:
Using LocalStateHolder now